pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -204,49 +260,97 @@ func (o *MustGatherOptions) Run(rsyncCmd *cobra.Command) error {
 		}()
 	}
 
-	// create pod
-	pod, err := o.Client.CoreV1().Pods(ns.Name).Create(o.newPod(o.NodeName))
-	if err != nil {
-		return err
+	// create pods
+	var pods []*corev1.Pod
+	for _, image := range o.Images {
+		pod, err := o.Client.CoreV1().Pods(ns.Name).Create(o.newPod(o.NodeName, image))
+		if err != nil {
+			return err
+		}
+		o.log("[%s] pod for plug-in image %s created", pod.Name, image)
+		pods = append(pods, pod)
 	}
 
-	// wait for gather container to be running (gather is running)
-	if err := o.waitForGatherContainerRunning(pod); err != nil {
-		return err
-	}
+	var wg sync.WaitGroup
+	wg.Add(len(pods))
+	errs := make(chan error, len(pods))
+	for _, pod := range pods {
+		go func(pod *corev1.Pod) {
+			defer wg.Done()
+
+			// wait for gather container to be running (gather is running)
+			if err := o.waitForGatherContainerRunning(pod); err != nil {
+				o.log("[%s] gather did not start: %s", pod.Name, err)
+				errs <- fmt.Errorf("gather did not start for pod %s: %s", pod.Name, err)
+				return
+			}
+			// stream gather container logs
+			if err := o.getGatherContainerLogs(pod); err != nil {
+				o.log("[%s] gather logs unavailable: %v", pod.Name, err)
+			}
 
-	// stream gather container logs
-	if err := o.getInitContainerLogs(pod); err != nil {
-		fmt.Fprintf(o.Out, "container logs unavailable: %v\n", err)
+			// wait for pod to be running (gather has completed)
+			o.log("[%s] waiting for gather to complete ", pod.Name)
+			if err := o.waitForPodRunning(pod); err != nil {
+				o.log("[%s] gather never finished: %v", pod.Name, err)
+				errs <- fmt.Errorf("gather never finished for pod %s: %s", pod.Name, err)
+				return
+			}
+
+			// copy the gathered files into the local destination dir
+			o.log("[%s] downloading gather output", pod.Name)
+			if err := o.copyFilesFromPod(pod, len(pods) > 1); err != nil {
+				o.log("[%s] gather output not downloaded: %v\n", pod.Name, err)
+				errs <- fmt.Errorf("unable to download output from pod %s: %s", pod.Name, err)
+				return
+			}
+		}(pod)
 	}
+	wg.Wait()
+	return aggregateErrorOrNil(errs)
+}
 
-	// wait for pod to be running (gather has completed)
-	if err := o.waitForPodRunning(pod); err != nil {
-		return err
+func aggregateErrorOrNil(errs <-chan error) error {
+	c := len(errs)
```

Comment:
You can replace this entire body with a simple for loop over channels, which will do exactly what you need:
```go
var arr []error 
for i := range errs {
    arr = append(arr, i)
}
return errors.NewAggregate(arr)
```
It will stop iterating when the channel is closed, and `NewAggregate` will return `nil` when passed an empty array.

Reply:
https://github.com/openshift/oc/pull/60